### PR TITLE
Introduction problem and user preferences fixes

### PR DESCRIPTION
**Pull request by coopdanger**

### CODE REVIEW

**Comment by coopdanger** [comment]
Incorporates fix suggested in #12

File: charter.md [hunk 1]
```diff
@@ -4,7 +4,9 @@ The More Instant Messaging Interoperability (MIMI) working group will specify th
...
 
-Achieving interoperability requires a solution for the introduction problem, i.e., the ability for a user in one application to take an identifier for a target user, determine the application in which that target should be reached, be granted permission to initiate communications, and be able to establish communications with the target. The working group will specify a solution to the introduction problem, together with best practice recommendations for functionality, configuration options, and other aspects. Express user preferences about discoverability and reachability must be respected. 
+Achieving interoperable messaging among MIMI-compliant services requires a solution for the introduction problem, i.e., given an identifier for a target user, a sending user should be able to establish communications with the target after the target user's service has been reached and the sender has been granted permission to initiate communications. The working group will specify a solution to the introduction problem, together with best practice recommendations for functionality, configuration options, and other aspects.
```
Comment (ekr) [inline charter.md:7]: This is better than the original text, but still does not clearly differentiate between:

1. Contacting a user when you know what service they are on
2. Discovering what service they are on.

There seemed to be relatively broad support for https://github.com/coopdanger/ietf-wg-mimi/pull/10, which tries to make this sharp. I don't care about whose text we used, but I'd like to understand what you thought was wrong with that so we can get to the right place.

File: charter.md [hunk 2]
```diff
@@ -4,7 +4,9 @@ The More Instant Messaging Interoperability (MIMI) working group will specify th
...
+
+Some messaging services today have features which define how a user is contacted by someone new. Some services require the user's approval while others allow inbound messages from any other user. This group's work will respect the use of these features and allow them to work the same when inbound messages come from an interoperating service.
```
Comment (ekr) [inline charter.md:9]: I don't know what "respect" means in this case.

File: charter.md (hunk 1 above)
Comment (coopdanger) [reply charter.md:7]: I think #10 implies that the sending user knows what service the receiving user is on. That is, when Alice wants to contact Bob whose phone number is 203-857-6309, Alice has to specify that she wants to contact Bob on WhatsApp or Signal or iMessage. I think this is suboptimal in terms of the amount of improvement to the user experience that interoperability would provide compared to the situation today. But I may be in the rough.

Comment (ekr) [reply charter.md:7]: I would say rather that it implies that there are two distinct functions (1) knowing what service that the user is on and (2) making the connection once that is known, and that the WG will work on (2) and may work on (1).  I agree that only doing (2) is less good than doing both, but I understood the sense of the room to be that that's what people wanted the charter to say.

Comment (coopdanger) [reply charter.md:7]: I think saying that the user will "take" the target identifier+associated application means that the user knows the associated application.

File: charter.md (hunk 2 above)
Comment (coopdanger) [reply charter.md:9]: I think the point is that if the user wants these features to continue working the same in the presence of interoperating services then that should be what happens.

File: charter.md (hunk 1 above)
Comment (ekr) [reply charter.md:7]: Yes. Hence the new text in PR#10 that addresses discovering it.

Comment (ekr) [reply charter.md:7]: I don't think the issue here is the term "introduction problem". Rather, it's whether the charter should take a position on whether discovering which service a user is on is in scope or not.

Comment (coopdanger) [reply charter.md:7]: A difference between #10 and Ted’s text (or my text) is that #10 puts something like SPIN out of scope. I don’t think it should be. The sentence in #10 about how the WG may work on a discovery solution is fine (and separate from the first sentence) IMO.

**Pull request closed without merging**